src/guides/v2.3/howdoi/checkout/checkout_custom_validation.md

```diff
@@ -0,0 +1,132 @@
+---
+layout: tutorial
+group: how-do-i
+subgroup:
+title: Add custom validation rules to checkout address form
+contributor_name: Ziffity
+contributor_link: https://www.ziffity.com/
+subtitle: Customize Checkout
+menu_order: 101
+level3_subgroup: checkout-tutorial
+functional_areas:
+  - Checkout
+---
+
+This topic describes how to add custom address validation to the Magento checkout.  By default it has own set of validations for the address fields in the checkout step. But sometimes you may want to add some custom validations to the different address fields. One example may be limiting the character limit in the firstname and lastname of the shipping and billing addresses.
+
+To change the existing address validations to the Magento checkout, do the following steps:
+
+1. [Create a new module](#create-module).
+1. [Create a di.xml file](#add-dixml-file).
+1. [Adding validation to the Shipping Address](#add-shipping-validation).
+1. [Adding validation to the Billing Address](#add-billing-validation).
+
+## Step 1: Create a new module {#create-module}
+
+[Create a new module](https://devdocs.magento.com/videos/fundamentals/create-a-new-module/) named `Vendor/AddressRestriction` and register it.
+
+## Step 2 Create a di.xml file {#add-dixml-file}
+
+Create the file `app/code/Vendor/AddressRestriction/etc/di.xml`:
+
+```php?start_inline=1
+<?xml version="1.0"?>
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:ObjectManager/etc/config.xsd">
+    <type name="Magento\Checkout\Block\Checkout\LayoutProcessor">
+        <plugin name="Vendor_AddressRestriction_Checkout" type="Vendor\AddressRestriction\Block\LayoutProcessor" sortOrder="100"/>
+    </type>
+</config>
+```
+
+## Step 3 Adding validation to the Shipping Address {#add-shipping-validation}
+
+Create the file `app/code/Vendor/AddressRestriction/Block/LayoutProcessor.php`;
+
+```php?start_inline=1
+<?php
+namespace Vendor\AddressRestriction\Block;
+
+class LayoutProcessor
+{
+    /**
+    * @param \Magento\Checkout\Block\Checkout\LayoutProcessor $subject
+    * @param array $jsLayout
+    * @return array
+    */
+    public function afterProcess(
+    \Magento\Checkout\Block\Checkout\LayoutProcessor $subject,
+    array $jsLayout
+    ) {
+        // Firstname
+        $jsLayout['components']['checkout']['children']['steps']['children']['shipping-step']['children']
```

Comment:
In my opinion it's much better to suggest using the [Array Manager](https://devdocs.magento.com/guides/v2.4/extension-dev-guide/framework/array-manager.html), which is much preferred way of accessing a chained array. As result we'll have a much cleaner solution and code.

Reply:
@eduard13 Thanks for your review and suggestion. I have updated the mentioned suggestion. Kindly look into that.

Reply:
@rajesh-mage, looks like you didn't get me right. I suggest using the Array Manager to access the array properties, please check the 1st example in the doc's article.

Please let me know if there are any concerns or questions.